manager/controllers/app/storage.go

```diff
@@ -6,168 +6,95 @@ package app
 import (
 	"context"
 
+	"emperror.dev/errors"
 	"github.com/go-logr/logr"
 	app "github.com/ibm/the-mesh-for-data/manager/apis/app/v1alpha1"
-	statusErr "k8s.io/apimachinery/pkg/api/errors"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
+	comv1alpha1 "github.com/IBM/dataset-lifecycle-framework/src/dataset-operator/pkg/apis/com/v1alpha1"
 	"github.com/ibm/the-mesh-for-data/manager/controllers/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 )
 
-// +kubebuilder:rbac:groups=app.m4d.ibm.com,resources=m4dbuckets,verbs=get;list;watch;update;
-// +kubebuilder:rbac:groups=app.m4d.ibm.com,resources=m4dbuckets/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=app.m4d.ibm.com,resources=m4dstorageaccounts,verbs=get;list;watch;update;
+// +kubebuilder:rbac:groups=com.ie.ibm.hpsys,resources=datasets,verbs=get;list;watch;create;update;patch;delete
 
-// FindAvailableBucket finds an available storage asset - in the relevant geo
-// TODO: Implement check for geo.  Needed for ingest
-func FindAvailableBucket(c client.Client, log logr.Logger, owner types.NamespacedName, id string, prefixBase string, canShare bool, geo string) *app.M4DBucket {
-	ctx := context.Background()
+// ProvisionInterface is an interface for managing Dataset resources
```

Comment:
Maybe call out in the comment that it's an interface for managing dynamically allocated dataset resources

Reply:
OK.